packages/invoices/src/OutboundShipment/DetailView/modals/TaxEdit.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import {
+  useTranslation,
+  NonNegativeNumberInputModal,
+  IconButton,
+  EditIcon,
+  useToggle,
+} from '@openmsupply-client/common';
+import { useUpdateOutboundTax } from '../../api';
+
+export const TaxEdit = ({ tax }: { tax: number }) => {
+  const { mutate } = useUpdateOutboundTax();
+  const modalController = useToggle();
+  const t = useTranslation('distribution');
+
+  return (
+    <>
+      <IconButton
+        icon={<EditIcon style={{ fontSize: 12, fill: 'none' }} />}
+        label={t('heading.edit-tax-rate')}
+        onClick={modalController.toggleOn}
+      />
+
+      {modalController.isOn && (
```

Comment:
unmounting whenever the modal is closed to reset states. I don't particularly like this pattern, but it seems easier than any alternatives

Reply:
sneaky - I might need to add a comment to remind me why this is here.
yeah, easier than alternatives, it's a neat way around the problem